src/components/tx/SignOrExecuteForm/index.tsx

```diff
@@ -34,7 +34,7 @@ const SignOrExecuteForm = (props: SignOrExecuteProps): ReactElement => {
   const willExecute = (props.onlyExecute || shouldExecute) && canExecute
 
   // Estimate gas limit
-  const { gasLimit, gasLimitError } = useGasLimit(safeTx)
+  const { gasLimit, gasLimitError } = useGasLimit(willExecute ? safeTx : undefined)
```

Comment:
Why this change? Gas limit is needed for simulation. If it's only estimated for executions, it should be moved to the ExecuteForm.

Reply:
This was a regression when we refactored this component. Moving it into the `ExecuteForm` would make the Simulation always fetch the latest block gas limit which is less accurate so I would suggest keeping it here for now.